addons/stock/models/stock_lot.py

```diff
@@ -127,15 +127,15 @@ def _compute_delivery_ids(self):
             if lot.product_id.tracking == 'serial' and lot.delivery_count > 0:
                 lot.last_delivery_partner_id = lot.delivery_ids.sorted(key=attrgetter('date_done'), reverse=True)[0].partner_id
 
-    @api.depends('quant_ids')
+    @api.depends('quant_ids', 'quant_ids.package_id')
     def _compute_single_location(self):
         for lot in self:
-            location_id = lot.quant_ids.filtered(lambda q: q.quantity > 0).mapped('location_id')
-            lot.location_id = location_id if len(location_id) == 1 else False
+            single_quant_id = lot.quant_ids.filtered(lambda q: q.quantity > 0)
+            lot.location_id = single_quant_id.location_id if len(single_quant_id) == 1 and not single_quant_id.package_id else False
```

Comment:
if we show those quants, we need to be able to move them, for that we need to remove the package otherwise we trigger a UserError in stock.move._action_done() due to having 2 quants in 2 locations in the same package before _merge_quants() has cleaned the quants. 

So we would a warning that the quant will be unpacked if they move it: either a confirmation pop-up or a red/yellow  banner. Since it was starting to add more complexity to the UX, we decided with THD to simply not show packed quants on a lot form.